harness/determined/cli/experiment.py

```diff
@@ -527,25 +530,33 @@ def kill_experiment(args: Namespace) -> None:
     print("Killed experiment {}".format(args.experiment_id))
 
 
-@authentication.required
-def wait(args: Namespace) -> None:
+def _wait(session: api.Session, experiment_id: int, polling_interval=5) -> None:
+    retry = urllib3.util.retry.Retry(
+        raise_on_status=False,
+        total=10,
+        status=10,
+        backoff_factor=2,
+        status_forcelist=[413, 429, 502, 503, 504],
+    )
+    session._max_retries = retry
```

Comment:
preface, not a python guru so please excuse dumb comments. if we mutate the session, will all calls after use the same `_max_retries`. and if that is the case, does it make sense to either do something more generic (`cli.setup_session(args, retry=True)` with some reasonable default retry) or not mutate it?

i'd don't think it matters here, exactly. i'm more thinking about this pattern going forward.

Reply:
> if we mutate the session, will all calls after use the same _max_retries

if I'm not mistaken this should just affect this single session. and we're not persisting any changes that would be shared with other session instances
https://github.com/hamidzr/determined/blob/4b66bf07fa501d7f3fe6fe1feb8be6ac810b834f/harness/determined/common/requests.py#L40

> if that is the case, does it make sense to either do something that makes this behavior more explicit (cli.setup_session(args, retry=True) with some reasonable default retry) or not mutate it?

with my last comment I don't think this is the case but what you're suggesting makes sense to me however if I I wasn't sure how I can test it. mainly because then I can't replace the bulk of session, authentication, and cert config  in my test code.

Reply:
I created a ticket for looking into default retries across all cli actions https://determinedai.atlassian.net/browse/DET-8684

Reply:
> if I'm not mistaken this should just affect this single session. and we're not persisting any changes that would be shared with other session instances

i agree it will affect just this session; my point was i see no reason to mutate the session so that everything gets retries halfway through some CLI action if the session is reused, but to just construct the session with retries. but i do think it is complex to think through (probably shouldn't retry _everything_)